lib/ret_web/channels/hub_channel.ex

```diff
@@ -0,0 +1,86 @@
+defmodule RetWeb.HubChannel do
+  @moduledoc "Ret Web Channel for Hubs"
+
+  use RetWeb, :channel
+
+  alias Ret.{Hub, Repo, SessionStat}
+
+  def join("hub:" <> hub_sid, _payload, socket) do
+    hub = Repo.get_by(Hub, hub_sid: hub_sid)
+    socket |> join_with_hub(hub)
+  end
+
+  def handle_in("events:entered", %{"initialOccupantCount" => occupant_count} = payload, socket) do
+    socket
+    |> handle_max_occupant_update(occupant_count)
+    |> handle_entered_event(payload)
+
+    {:noreply, socket}
+  end
+
+  def handle_in("events:entered", payload, socket) do
+    handle_entered_event(socket, payload)
+
+    {:noreply, socket}
+  end
+
+  def handle_in(_message, _payload, socket) do
+    {:noreply, socket}
+  end
+
+  def terminate(_reason, socket) do
+    socket
+    |> SessionStat.stat_query_for_socket()
+    |> Repo.update_all(set: [ended_at: DateTime.utc_now()])
+
+    :ok
+  end
+
+  defp join_with_hub(socket, %Hub{} = hub) do
+    with socket <- assign(socket, :hub_sid, hub.hub_sid),
+         response <- RetWeb.Api.V1.HubView.render("show.json", %{hub: hub}) do
+      existing_stat_count =
+        socket
+        |> SessionStat.stat_query_for_socket()
+        |> Repo.all()
+        |> length
+
+      unless existing_stat_count > 0 do
+        with session_id <- socket.assigns.session_id,
+             started_at <- socket.assigns.started_at,
+             stat_attrs <- %{session_id: session_id, started_at: started_at},
+             changeset <- %SessionStat{} |> SessionStat.changeset(stat_attrs) do
+          Repo.insert(changeset)
+        end
+      end
+
+      {:ok, response, socket}
+    end
+  end
+
+  defp join_with_hub(socket, nil) do
```

Comment:
`join_with_hub(_, nil)` or `join_with_hub(nil, _)` if you take my other suggestion above